File: frontend/src/lib/events.ts
// Wails event name constants — must stay in sync with internal/events/events.go

export const TaskCreated = 'task:created'
export const TaskUpdated = 'task:updated'
export const TaskDeleted = 'task:deleted'

export const AgentStatePrefix = 'agent:state:'
export const AgentOutputPrefix = 'agent:output:'
export const AgentErrorPrefix = 'agent:error:'

export const OrchestratorState = 'orchestrator:state'
export const ReviewsUpdated = 'reviews:updated'
export const Notification = 'notification'
export const AppQuitConfirm = 'app:quit-confirm'

export const agentState = (id: string) => `${AgentStatePrefix}${id}`
export const agentOutput = (id: string) => `${AgentOutputPrefix}${id}`
export const agentError = (id: string) => `${AgentErrorPrefix}${id}`

File: internal/events/events.go
// Package events defines Wails event name constants shared across the app.
package events

const (
	// Task lifecycle events (emitted by watcher).
	TaskCreated = "task:created"
	TaskUpdated = "task:updated"
	TaskDeleted = "task:deleted"

	// Agent events — prefix only; append ":"+agentID to form full event name.
	AgentStatePrefix  = "agent:state:"
	AgentOutputPrefix = "agent:output:"
	AgentErrorPrefix  = "agent:error:"
	AgentStuckPrefix  = "agent:stuck:"

	// Orchestrator events.
	OrchestratorState = "orchestrator:state"

	// Review/PR events.
	ReviewsUpdated = "reviews:updated"

	// Notification events.
	Notification = "notification"

	// App lifecycle events.
	AppQuitConfirm = "app:quit-confirm"
)

// AgentState returns the agent state event name for the given agent ID.
func AgentState(id string) string { return AgentStatePrefix + id }

// AgentOutput returns the agent output event name for the given agent ID.
func AgentOutput(id string) string { return AgentOutputPrefix + id }

// AgentError returns the agent error event name for the given agent ID.
func AgentError(id string) string { return AgentErrorPrefix + id }

// AgentStuck returns the agent stuck event name for the given agent ID.
func AgentStuck(id string) string { return AgentStuckPrefix + id }
